### ISSUE TITLE
S1075 FN: Mark list pattern test cases as compliant

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6380]
### Description

FN for S1075 (UriShouldNotBeHardcoded).

Using C#11 list patterns, the rule is not raising an issue.

### Repro steps

```csharp
    public bool ListPattern(string[] uris) =>
        uris is ["""\\my-network-drive\folder\file.txt"""]; // FN

```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
